packages/manager/apps/pci-databases-analytics/src/pages/services/[serviceId]/users/_components/ToggleAcl.component.tsx

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Label } from '@/components/ui/label';
+import { Switch } from '@/components/ui/switch';
+import { useServiceData } from '../../Service.context';
+import { useToast } from '@/components/ui/use-toast';
+import { useEditService } from '@/hooks/api/database/service/useEditService.hook';
+import * as database from '@/types/cloud/project/database';
+
+const ToggleAcl = () => {
+  const { t } = useTranslation(
+    'pci-databases-analytics/services/service/users',
+  );
+  const { projectId, service, serviceQuery } = useServiceData<
+    database.opensearch.Service
+  >();
+  const aclsEnabled = !!('aclsEnabled' in service && service.aclsEnabled);
+  const [switchState, setSwitchState] = useState(aclsEnabled);
+
+  const toast = useToast();
+  const { editService, isPending } = useEditService<
+    database.opensearch.Service
+  >({
+    onError: (err) => {
+      toast.toast({
+        title: t('toggleAclErrorTitle'),
+        variant: 'destructive',
+        description: err.response.data.message,
+      });
+      setSwitchState(aclsEnabled);
+    },
+    onEditSuccess: (updatedService) => {
+      toast.toast({
+        title: t('toggleAclSuccessTitle'),
+        description: updatedService.aclsEnabled
+          ? t('toggleAclEnabledSuccessDescription')
+          : t('toggleAclADisabledSuccessDescription'),
+      });
+      serviceQuery.refetch();
+    },
+  });
+
+  useEffect(() => {
+    setSwitchState(service.aclsEnabled);
+  }, [service.aclsEnabled]);
```

Comment:
As per the React docs about `useEffect`, there is no need here because it should be computed at render